packages/twenty-server/src/engine/core-modules/user/user.resolver.ts

```diff
@@ -113,4 +124,24 @@ export class UserResolver {
     // Proceed with user deletion
     return this.userService.deleteUser(userId);
   }
+
+  @ResolveField(() => Boolean, { nullable: true })
+  async skipSyncEmail(@AuthUser() user: User): Promise<boolean | null> {
+    const connectedAccounts =
+      await this.connectedAccountRepository.getAllByUserId(
+        user.id,
+        user.defaultWorkspaceId,
+      );
+
+    if (connectedAccounts?.length) {
+      return true;
+    }
+    const skipSyncEmail = await this.keyValuePairService.get(
```

Comment:
can we introduce another layer on top of keyValuePair which is not a product concept.
Something like userState? then I would actually return the whole userState and attacht it to the user